protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java

```diff
@@ -1206,6 +1208,23 @@ public KeyExchangeResult getSharedSecret(byte[] message, Composite ownState, Key
       case Const.ASYMKEX2048_ALG_NAME:
       case Const.ASYMKEX3072_ALG_NAME:
         return getAsymkexSharedSecret(message, ownState, decryptionKey);
+      case DiffieHellman.DH14_ALG_NAME:
+      case DiffieHellman.DH15_ALG_NAME:
+        // All owner states have our message as element 0: restore our keyExchange
+        // from the KE state, which is element 2 in the owner state.
```

Comment:
very picky here - comment is correct but was confusing at first. Adding "(THIRD_KEY)" after "element 2" would make it clearer.

Reply:
Fixed.